### PR TITLE
ENH: implement Timedelta._as_unit

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/scalar/timedelta/test_timedelta.py [hunk 1]
```diff
@@ -24,6 +25,80 @@
...
+
+        res = td._as_unit("s")
+        assert res.value == td.value // 1_000_000_000
+        assert res._reso == td._reso - 3
+
+        rt = res._as_unit("ns")
+        assert rt.value == td.value
+        assert rt._reso == td._reso
+
+    def test_as_unit_overflows(self):
```
Comment (mroeschke) [inline pandas/tests/scalar/timedelta/test_timedelta.py:58]: Just confirming aloud, based on the multiplication operation we can't get to a state where underflowing occurs (and therefore can't test that)?

File: pandas/tests/scalar/timedelta/test_timedelta.py [hunk 2]
```diff
@@ -24,6 +25,80 @@
...
+        assert res == expected
+
+        assert res._reso == 8
+        assert res.value == 1
+
+        with pytest.raises(ValueError, match="Cannot losslessly convert units"):
+            td._as_unit("ms", round_ok=False)
+
+    def test_as_unit_non_nano(self):
+        # case where we are going neither to nor from nano
```
Comment (mroeschke) [inline pandas/tests/scalar/timedelta/test_timedelta.py:85]: Not really sure what `neither to nor` means (if it's an idiom)

**Review by mroeschke — COMMENTED** [review]
So `_as_unit` will be used separately from `_as_reso` in the future?

**Comment by jbrockmendel** [comment]
> So _as_unit will be used separately from _as_reso in the future?

Yes _as_unit is the one that will eventually be user-facing.  Just didnt want to make it public yet in case we want to change the name

**Review by mroeschke — APPROVED** [review]

**Pull request merged**